backend/src/main/java/org/cryptomator/hub/validation/ValidName.java

```diff
@@ -0,0 +1,26 @@
+package org.cryptomator.hub.validation;
+
+import javax.validation.Constraint;
+import javax.validation.Payload;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.Pattern;
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.*;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Pattern(regexp = "(?Ui)[-_\\w\\s]+")
+@NotBlank
+@Target({METHOD, FIELD, ANNOTATION_TYPE, TYPE_USE, PARAMETER})
+@Retention(RUNTIME)
+@Constraint(validatedBy = {})
+@Documented
+public @interface ValidName {
```

Comment:
Imho this is still too specific. I would like you to only create custom annotations for "types" such as JWEs, UUIDs, etc, not for "meanings" such as names, descriptions, comments, ... In the latter cases, just apply the `@Pattern` annotation directly where needed.

(Also, it doesn't match `Nick O’Neill`.)

Reply:
See https://github.com/cryptomator/hub/pull/154/commits/e1230648ffa6ddcdf4043565e1f1ee2f091249e4